core/src/test/java/com/gentics/mesh/core/field/s3binary/S3BinaryFieldEndpointTest.java

```diff
@@ -0,0 +1,269 @@
+        MeshBinaryResponse response = call(() -> client().downloadBinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "en", "s3",
+                new ImageManipulationParametersImpl().setWidth(100)));
+        try {
+            byte[] downloadBytes = IOUtils.toByteArray(response.getStream());
+            InputStream in = new ByteArrayInputStream(downloadBytes);
+            buf = ImageIO.read(in);
+        } catch (IOException ioException) {
+            fail();
+        }
+        assertEquals(100, buf.getWidth());
+        assertNotNull(response);
+        assertEquals("image/jpeg", response.getContentType());
+        response.close();
+    }
+
+    @Test
+    public void testTransformS3BinarySuccessful() {
+        S3RestResponse s3RestResponse = new S3RestResponse();
+        s3RestResponse.setVersion("1");
+        NodeResponse s3binaryNode = createNodeWithField();
+        //creating
+        call(() -> client().updateNodeS3BinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "s3", uploadBody));
+        //uploading
+        File tempFile = createTempFile();
+        s3BinaryStorage().createBucket("test-bucket").blockingGet();
+        s3BinaryStorage().createBucket("test-cache-bucket").blockingGet();
+        s3BinaryStorage().uploadFile("test-bucket", s3binaryNode.getUuid() + "/s3", tempFile, false).blockingGet();
+        client().extractMetadataNodeS3BinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "s3", metadataBody).blockingGet();
+        NodeResponse call = call(() -> client().transformNodeBinaryField(PROJECT_NAME, s3binaryNode.getUuid(), "en", "1.0", "s3", new ImageManipulationParametersImpl().setWidth(250)));
+        assertNotNull(call);
+        assertEquals(250, call.getFields().getS3BinaryField(FIELD_NAME).getWidth().intValue());
+    }
+
+    @Test
+    public void testUploadWorking() {
+        S3RestResponse s3RestResponse = new S3RestResponse();
+        s3RestResponse.setVersion("1");
+        NodeResponse s3binaryNode = createNodeWithField();
+        S3RestResponse s3RestResponse1 = null;
+        //creating
+        try (Tx tx = tx()) {
+            call(() -> client().updateNodeS3BinaryField(
+                    PROJECT_NAME,
+                    s3binaryNode.getUuid(),
+                    "s3",
+                    uploadBody
+            ));
+            //uploading
+            File tempFile = createTempFile();
+            s3BinaryStorage().createBucket("test-bucket").blockingGet();
+            s3RestResponse1 = s3BinaryStorage().uploadFile("test-bucket", s3binaryNode.getUuid() + "/s3", tempFile, false).blockingGet();
+        }
+        Boolean doesObjectExists = s3BinaryStorage().exists("test-bucket", s3binaryNode.getUuid() + "/s3").blockingGet();
+        assertTrue(doesObjectExists);
+        assertNotNull(s3RestResponse1);
+    }
+
+    @Test
+    public void testUploadEmptyFilename() {
+        NodeResponse s3binaryNode = createNodeWithField();
+        String body = "{\n" +
```

Comment:
Same here - should be no artificially constructed body as a MeshRestClient parameter.